flask/app/gene_viewer.py

```diff
@@ -0,0 +1,248 @@
+    enum_validate,
+    filter_in_enum_or_abort,
+    filter_updated_or_abort,
+    mixin,
+    paged,
+    transaction_or_abort,
+)
+
+
+gene_viewer_blueprint = Blueprint(
+    "gene_viewer",
+    __name__,
+)
+
+
+def get_variant_wise_df(statement: Select, session: Engine):
+
+    df = pd.read_sql(statement, session)
+
+    app.logger.debug(df.head(3))
+
+    df = df[
+        [
+            "position",
+            "reference_allele",
+            "alt_allele",
+            "variation",
+            "refseq_change",
+            "depth",
+            "gene",
+            "conserved_in_20_mammals",
+            "sift_score",
+            "polyphen_score",
+            "cadd_score",
+            "gnomad_af",
+            "zygosity",
+            "burden",
+            "alt_depths",
+            "dataset_id",
+            "participant_id",
+            "participant_codename",
+            "family_id",
+            "family_codename",
+        ]
+    ]
+    df = df.loc[:, ~df.columns.duplicated()]
+    # some columns are duplicated eg. dataset_id, is there a way to query so that this doesn't happen?
+
+    df = df[~df["zygosity"].str.contains("-|Insufficient")]
+
+    df = df.astype(str)
+    df = df.groupby(["position", "reference_allele", "alt_allele"]).agg(
+        {
+            "variation": "first",
+            "refseq_change": "first",
+            "depth": list,
+            "gene": "first",
+            "conserved_in_20_mammals": "first",
+            "sift_score": "first",
+            "polyphen_score": "first",
+            "cadd_score": "first",
+            "gnomad_af": "first",
+            "zygosity": list,
+            "burden": list,
+            "alt_depths": list,
+            "dataset_id": list,
+            "participant_id": list,
+            "participant_codename": list,
+            "family_id": lambda x: set(x),
+            "family_codename": lambda x: set(x),
+        },
+        axis="columns",
+    )
+
+    df["frequency"] = df["participant_codename"].str.len()
+
+    for col in [
+        "zygosity",
+        "burden",
+        "alt_depths",
+        "depth",
+        "participant_codename",
+        "family_codename",
+        "dataset_id",
+        "participant_id",
+        "family_id",
+    ]:
+        df[col] = df[col].apply(lambda g: "; ".join(g))
+    return df
+
+
+@gene_viewer_blueprint.route("/api/summary/variants", methods=["GET"])
+@login_required
+def variant_summary():
+
+    app.logger.debug("Retrieving user id..")
+
+    if app.config.get("LOGIN_DISABLED") or current_user.is_admin:
+        user_id = request.args.get("user")
+        app.logger.debug("User is admin with ID '%s'", user_id)
+    else:
+        user_id = current_user.user_id
+        app.logger.debug("User is regular with ID '%s'", user_id)
+
+    if request.headers["Accept"] not in ["application/json", "text/csv"]:
+        app.logger.error("Unsupported HTTP accept")
+        abort(406, description="Unsupported HTTP accept")
```

Comment:
The description should indicate the supported types, but really I think we should just special-case `text/csv` and return JSON by default (or the other way around) since usually the value of this header is some variant of `*/*`